external/storm-jdbc/src/test/java/org/apache/storm/jdbc/common/JdbcClientTest.java

```diff
@@ -80,6 +85,24 @@ public void testInsertAndSelect() {
         Assert.assertEquals(rows, selectedRows);
     }
 
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void testInsertConnectionError() {
+
+        ConnectionProvider connectionProvider = new MockConnectionProvider(null);
+        this.client = new JdbcClient(connectionProvider, 60);
+
+        List<Column> row = createRow(1, "frank");
+        List<List<Column>> rows  = new ArrayList<List<Column>>();
+        rows.add(row);
+        String query  = "insert into user_details values(?,?,?)";
+
+        thrown.expect(MultipleFailureException.class);
```

Comment:
It's thrown from closeConnection via throw new IllegalStateException(finalException). If an exception is embedded into another exception, it will be thrown as MultipleFailureException.